pandas/io/excel/_calaminereader.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+from datetime import (
+    date,
+    datetime,
+    time,
+)
+from typing import (
+    TYPE_CHECKING,
+    Union,
+)
+
+from pandas.compat._optional import import_optional_dependency
+from pandas.util._decorators import doc
+
+import pandas as pd
+from pandas.core.shared_docs import _shared_docs
+
+from pandas.io.excel._base import BaseExcelReader
+
+if TYPE_CHECKING:
+    from pandas._typing import (
+        FilePath,
+        ReadBuffer,
+        Scalar,
+        StorageOptions,
+    )
+
+_CellValueT = Union[int, float, str, bool, time, date, datetime]
+
+
+class CalamineExcelReader(BaseExcelReader):
+    @doc(storage_options=_shared_docs["storage_options"])
+    def __init__(
+        self,
+        filepath_or_buffer: FilePath | ReadBuffer[bytes],
+        storage_options: StorageOptions = None,
+    ) -> None:
+        """
+        Reader using calamine engine (xlsx/xls/xlsb/ods).
+
+        Parameters
+        ----------
+        filepath_or_buffer : str, path to be parsed or
+            an open readable stream.
+        {storage_options}
+        """
+        import_optional_dependency("python_calamine")
+        super().__init__(filepath_or_buffer, storage_options=storage_options)
+
+    @property
+    def _workbook_class(self):
+        from python_calamine import CalamineWorkbook
+
+        return CalamineWorkbook
+
+    def load_workbook(self, filepath_or_buffer: FilePath | ReadBuffer[bytes]):
+        from python_calamine import load_workbook
+
+        return load_workbook(filepath_or_buffer)  # type: ignore[arg-type]
```

Comment:
Can you advise what the mypy errors are for this and the subsequent ones? Not necessarily a blocker but surprised to see these

Reply:
pyright:
```
  pandas/io/excel/_calamine.py:60:30 - error: Argument of type "FilePath | ReadBuffer[bytes]" cannot be assigned to parameter "path_or_filelike" of type "str | PathLike | ReadBuffer" in function "load_workbook"
    Type "FilePath | ReadBuffer[bytes]" cannot be assigned to type "str | PathLike | ReadBuffer"
      Type "ReadBuffer[bytes]" cannot be assigned to type "str | PathLike | ReadBuffer"
        "ReadBuffer[bytes]" is incompatible with "str"
        "ReadBuffer[bytes]" is incompatible with protocol "PathLike"
          "__fspath__" is not present
        "ReadBuffer[bytes]" is incompatible with protocol "ReadBuffer"
          "seek" is an incompatible type
            Type "(__offset: int, __whence: int = ..., /) -> int" cannot be assigned to type "() -> int" (reportGeneralTypeIssues)
```

mypy:
```
pandas/io/excel/_calamine.py:60: error: Argument 1 to "load_workbook" has incompatible type "Union[Union[str, PathLike[str]], pandas._typing.ReadBuffer[bytes]]"; expected "Union[str, PathLike[Any], python_calamine._python_calamine.ReadBuffer]"  [arg-type]
```